src/rofiles-fuse/main.c

```diff
@@ -385,7 +385,12 @@ do_open (const char *path, mode_t mode, struct fuse_file_info *finfo)
       if (opt_copyup)
         {
           (void) close (fd);
-          fd = openat (basefd, path, finfo->flags & ~O_TRUNC, mode);
+          /* Note that unlike the initial open, we will pass through
+           * O_TRUNC.  More ideally in this copyup case we'd avoid copying
+           * the whole file in the first place, but eh.  It's not like we're
+           * high performance anyways.
+           */
+          fd = openat (basefd, path, finfo->flags & ~(O_EXCL|O_CREAT), mode);
```

Comment:
Why are we filtering out `O_EXCL` and `O_CREAT` here? This is something that should've been caught by the first `openat`, right? Or is this just to not have a TOCTOU race on the file existing?

Reply:
We're internally reopening the file we just created.  So if the user provided `O_EXCL` we need to filter that out.  We might as well drop `O_CREAT` too.